components/builder-depot/src/lib.rs

```diff
@@ -89,7 +89,7 @@ impl DepotUtil for config::Config {
     where
         T: Identifiable,
     {
-        let file = self.archive_path(ident, target);
+        let file = self.archive_name(ident, target);
         match fs::metadata(&file) {
```

Comment:
Is the metadata check deterministic, since the file now no longer a fully qualified path?

Reply:
I _think_ so since it's now only ever called in the context of being within a `/pkgs/tempfile`

Reply:
I wonder if it's safer to pass in the parent path and do a fully qualified check.. or remove the metadata check (as it gets checked or will fail later anyways if the file isn't there).

Reply:
We can't know what the parent path is anymore as the parent path is generated dynamically by the request. We just know that we will be checking a hartfile at `/pkgs/<some_tempdir>/`

Reply:
Actually I take this back, it looks like after the refactor that function never actually gets called.